R/plot_fundamental_diagram.R

```diff
@@ -0,0 +1,328 @@
+#' Plot three fundamentals diagrams
+#'
+#'
+#' @param enriched_data enriched data.frame containing all the data for all your sensors
+#' @param date_range Date vector. example: c('2021-01-01','2022-01-01'). Full period if NULL (default).
+#' @param segments Character vector. Selected road segment, all if NULL (default).
+#' @param weekday_choice weekday Character vector. Weekday choosen. Default to the all week.
+#' @param hour_choice Integer vector. Hours choosen, default to the all day.
+#' @param vacation_choice Character vector. Selected vacation. Full period by default (NULL).
+#' @param holiday_choice Boolean. Selected holiday.  Full period by default (NULL).
+#' @param direction_choice Character vector. Selected direction ('rgt','lft'). Both by default (NULL)
+#'
+#' @return enriched_data
+#' @export
+#'
+#' @import dplyr
+#' @import lubridate
+#' @import ggplot2
+#'
+#' @examples
+#' plot_fundamental_diagramm(traffic)
+#' plot_fundamental_diagramm(traffic,
+#'   date_range = c('2022-07-01','2022-09-01'),
+#'   weekday_choice = c('monday','friday','sunday'),
+#'   hour_choice = c(1,5,10,14,21),
+#'   holiday_choice = TRUE,
+#'   segments = 'RteVitre-06',
+#'   direction_choice = 'rgt')
+plot_fundamental_diagramm <-function(enriched_data,
+                                      date_range = NULL,
+                                      weekday_choice = NULL,
+                                      hour_choice = NULL,
+                                      vacation_choice=NULL,
+                                      holiday_choice=NULL,
+                                      segments = NULL,
+                                      direction_choice=NULL)
+
+{
+  #Filter the user's demand
+  enriched_data<-filter_demand_user(enriched_data,segments,date_range,weekday_choice,hour_choice,
+                                    vacation_choice,holiday_choice)
+
+  id_seg<-unique(enriched_data$segment_id)
+  direction_choice<-tolower(direction_choice)
+
+  #Plot the fundamental diagramm for each sensor
+  for (id in id_seg)
+  {
+    df_seg<-enriched_data[enriched_data$segment_id==id,]
+    if(df_seg$Version[1]==2 & "speed_hist_car_lft" %in% colnames(df_seg)){df_seg<-recons_v85(df_seg)}
+    df_seg<-retrieve_missing_data(df_seg)
+
+    if(!("speed_hist_car_lft" %in% colnames(df_seg)))
+    {
+      df_seg<-calculate_axes(df_seg,1,NULL)
+      plot_diagramm(df_seg,1,NULL)
+    }
+
+    else
+    {
+      if(!is.null(direction_choice))
+      {
+        if(direction_choice=='lft')
```

Comment:
Is it possible to put directly :

```
f_seg<-calculate_axes(df_seg,2,direction_choice)
plot_diagramm(df_seg,2,direction_choice)
```

without the conditions `if(direction_choice=='lft')` ?